packages/components/src/ui/styles/README.md

```diff
@@ -0,0 +1,61 @@
+# styles
+
+## The `ui` object
+
+The primary API that `styles` exposes is the `ui` object. It contains the following properties:
+
+- `get`: Retrieves a pre-defined set of CSS Custom Properties. See [Theme Configuration](#theme-configuration) below.
+- `tokens`: A pre-defined set of theme variables, accessible via the frozen `tokens` object. See [Theme Tokens](#theme-tokens) below.
+- Various mixins. See [Mixins](#mixins) below.
+
+## Theme Configuration
+
+Each of the configs declares overrides for the theme variables. The overall list of available theme variables is declared in [`styles/theme/config.js`](./theme/config.js). Individual overrides can be found in [`styles/theme/dark-mode-config.js`](./theme/dark-mode-config.js), [`styles/theme/high-contrast-mode-config.js`](./theme/high-contrast-mode-config.js), and [`styles/theme/dark-high-contrast-mode-config.js`](./theme/dark-high-contrast-mode-config.js).
+
+Variables added to the overarching config should align with variables declared in the `base-styles` package.
+
+These variables are accessed via the `ui.get` helper exported from `styles` and are exposed as CSS Custom Properties.
+
+## Theme Tokens
+
+Tokens are similar to variables except that they are not exposed as CSS Custom Properties. This is to avoid introducing too many new public APIs. Instead, we simply access them off of the `ui.tokens` object as you would any normal JavaScript object. `ui.tokens` is a _frozen_ object so runtime changes are no allowed to it, updates to it must be constant and done in the [`styles/tokens.js`](./tokens.js) module.
```

Comment:
So this does remove the possibility to implement the existing G2 themes right? since themes can't change these variables?

just to clarify for me, this is fine as a start but this is where I think we potentially need a `ThemeProvider` that relies on "context" in order to support this kind of themes, and at that point, it becomes clear that there's no much difference between "tokens" and "configuration" since both could be define similary, the value of the configs can still be "CSS variables" while the values of the "tokens" can be just static values.

Reply:
Personally I think we should revert back to the existing solution and mark all the variables introduced by the new style system as `unstable` by prefixing them with `--wp-unstable-token` or something like that so that no one depends on them with the expectation that they won't change from under them.

I think this is best because it aligns with the original vision of the style system as relying solely on CSS custom properties, allowing theming through the original theme provider without introducing React context into the question which is a huge performance gain (if you're calling `useTheme` from every component in the tree this creates a significant performance hit that can be avoided just by using CSS custom properties).

What do you think, can we go back to the original implementation (with less variables of course) if we just mark the variables as unstable so we're not introducing a maintanence nightmare scenario?

Reply:
> What do you think, can we go back to the original implementation (with less variables of course) if we just mark the variables as unstable so we're not introducing a maintanence nightmare scenario?

I don't mind that personally yeah but I guess right now there's a big gap between what a "theme" is for G2 (hundreds of variables) and what a "theme" should be in my mind (a dozen variables at best). Do you think reconciling the two is  possible?

Reply:
I mean we've already removed the hundreds of variables and paired it down to just a few dozen. I think we should expand our definition of theme to include spacing and some other things important to the "fit" and "feel" of the components that could be really valuable to be manipulated by a consumer through the ThemeProvider API... but the primary concern for themes is colors and fonts, which I think we're achieved now.

If we could move the `tokens` back into the regular theme now that would solve this concern for me.

What do you percieve to be the gap at this point in the definition of what a theme is, now that we've removed almost all the variables?

Reply:
What we have now still feels a bit too much for me. I think G2's original goal was trying to allow "any theme" for any kind of design while I feel a WP focused component library should be more opinionated while allowing some  customizations. It is important to think of a "theme" in this context as a "theme" for the entire WP-Admin and not just the component library.

In a lot of ways G2 was built as multi-purpose component library for any kind of JS frontends, similar to the number of components libraries that exist in the wild. This vision is where I feel there's a contradiction with what we want to achieve: a coherent component library for WordPress. So opinions are fine and even good and customization should be minimal IMO.

Based on this, for me the theme should be:

 - A main color
 - Potentially a secondary color
 - A grid unit (space)
 
a font family and font size, maybe but even that I don't think it's necessary for us (at least not at first).

I'm not sure I understand what  you mean by "fit"/"feel".

Reply:
Also what makes me uneasy about what G2 does at the moment is that it was develop as its own project and had to make a lot of decisions with designers from WordPress being involved in these discussions about what "theme" we want to support and what should be opinionated.

Reply:
By fit and feel I just mean spacing, border radius, border colors, background colors for elements that sit on top the background surface, intermediary colors like modal backdrop colors, animation variables (which are especially useful for automatically handling reduced motion instead of having to duplicate reduced motion handling across all components... or removing animations altogether if that's the target use-case needs it)... these are just some of the things I think fall into the fit and feel of a component system's design system or theme.

I don't think we need all 300 or whatever colors we had before, but I do think some of the things we've dropped are important to maintain for a component system to be customizable.

Say for example the WooCommerce plugin wanted to use WordPress/components, they'd need to be able to customize almost all of these things for wp/components to fall inline with their pre-existing design language.

Reply:
WooCommerce plugin should customize the library to included its branding but it shouldn't be a departure from what the core styles are. This is where the different in vision stands, in the level of customizable things we want to have. I know @jasmussen has thoughts on this as well. It's clear to me for instance that the level of customizability G2 has currently is just way too much.

Reply:
There is an important balance to strike regarding how much is customizable. Too little and it might be a pain point, too much and it risks lacking character and direction. Ultimately that isn't something to decide completely, but I find it easier to start with less and add when it becomes necessary, rather than start with total customizability and realize we need to reel it in.

Reply:
Gotcha. If it's currently way too much then that's just how it is, let me know how best to pare it down and I'm happy to do so.

Personally I have no vision for this library, I just want it integrated so we can take advantage of the CSS-in-JS offerings it provides. I'm not married to any of the implementation details aside from ones I think give us a performance or usability edge over using raw Emotion.

The way this style system handles themes is _not_ one of those things (aside from the performance aspect I pointed out of not wanting to use React context to pass around theme variables, that I think we should keep in CSS custom properties with the `unstable` prefix).